docs/guides/universal-receiver-delegate/create-custom-urd-2.md

```diff
@@ -0,0 +1,294 @@
+
+```solidity title="contracts/MockContract.sol"
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.9;
+import {UniversalProfile} from '@lukso/lsp-smart-contracts/contracts/UniversalProfile.sol';
+```
+
+## Deploy our custom URD
+
+As we have 2 different methods with specific requirements, we will provide both set of code. Comment the code that doesn't match your choice.
+
+Now we can create the deploy scripts that will connect all the pieces together. This script will:
+
+- Connect to our UP
+- Deploy the custom URD as our UP
+- Register it as a URD in our UP
+
+and, depending on the method selected, it will also:
+
+- grants permission to the URD to call the UP (SUPER_CALL + REENTRANT) - Method 1
+- calls `authorizeOperator()` on the LSP7 token to authorize the URD to spend token on the UP's behalf - Method 2
+
+Create a file called `deployLSP1.ts` in your `scripts/` folder with the following content:
+
+```ts title="scripts/deployLSP1.ts"
+import hre from 'hardhat';
+import { ethers } from 'hardhat';
+import * as dotenv from 'dotenv';
+import {
+  ERC725YDataKeys,
+  LSP1_TYPE_IDS,
+  PERMISSIONS,
+} from '@lukso/lsp-smart-contracts';
+
+// load env vars
+dotenv.config();
+
+// You can update the value of the allowed LSP7 token
+const contractsAddr = ['0x63890ea231c6e966142288d805b9f9de7e0e5927'];
+
+// Update those values in the .env file
+const { UP_ADDR, PRIVATE_KEY, UP_RECEIVER, PERCENTAGE } = process.env;
+
+/**
+ * In this script, we will:
+ * - deploy our URD implementation (LSP1URDForwarder.sol)
+ * - setDataBatch on the UP to register URD implementation + permission for URD contract
+ */
+async function main() {
+  // ----------
+  // BASE SETUP
+  // ----------
+
+  // setup provider
+  const provider = new ethers.JsonRpcProvider(
+    'https://rpc.testnet.lukso.network',
+  );
+  // setup signer (the browser extension controller)
+  const signer = new ethers.Wallet(PRIVATE_KEY as string, provider);
+  // load the associated UP
+  const UP = await ethers.getContractAt('UniversalProfile', UP_ADDR as string);
+  console.log('🔑 EOA: ', signer.address);
+  console.log('🆙 Universal Profile: ', await UP.getAddress());
+
+  // ----------
+  // DEPLOY URD
+  // ----------
+
+  console.log('⏳ Deploying the custom URD');
+  const CustomURDBytecode =
+    hre.artifacts.readArtifactSync('LSP1URDForwarder').bytecode;
+
+  // we need to encode the constructor parameters and add them to the contract bytecode
+  const abiCoder = new ethers.AbiCoder();
+  const params = abiCoder
+    .encode(
+      ['address', 'uint256', 'address[]'],
+      [UP_RECEIVER as string, PERCENTAGE as string, contractsAddr],
+    )
+    .slice(2);
+  const fullBytecode = CustomURDBytecode + params;
+
+  // get the address of the contract that will be created
+  const CustomURDAddress = await UP.connect(signer).execute.staticCall(
+    1,
```

Comment:
Let's use the constants for clarity. To be imported `OPERATIONS_TYPE`

```suggestion
    OPERATIONS_TYPE.CREATE,
```

Reply:
done